runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkPipelineRunner.java

```diff
@@ -0,0 +1,212 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Launch a Beam pipeline via external SDK driver program in the Flink {@link OptimizerPlanEnvironment}.
+ *
+ * <p>This entry point can be used to execute an external program to launch a Beam pipeline within
+ * the Flink {@link OptimizerPlanEnvironment}, which is present when running a job via the REST
+ * API.
+ *
+ * <p>Designed for non-interactive Flink REST client and container with the Beam job server jar
+ * and SDK client (for example when using the FlinkK8sOperator).
+ *
+ * <p>Eliminates the need to build jar files with materialized pipeline protos offline. Allows
+ * the driver program to access actual execution environment and services, on par with code
+ * executed by SDK workers.
+ *
+ * <p>The entry point starts the job server and provides the endpoint to the the driver program.
+ *
+ * <p>The driver program constructs the Beam pipeline and submits it to the job service.
+ *
+ * <p>The job service submits the pipeline into the plan environment and returns the "detached"
+ * status to the driver program.
+ *
+ * <p>Upon completion of the driver program, the entry points signals completion of job construction
+ * to the environment and terminates.
+ *
+ * <p>Finally Flink launches the job.
+ */
+public class LyftFlinkPipelineRunner {
+  private static final Logger LOG = LoggerFactory.getLogger(LyftFlinkPipelineRunner.class);
+  private static String DRIVER_CMD_FLAGS = "--job_endpoint=localhost:%s";
+
+  private final String driverCmd;
+  private FlinkJobServerDriver driver = null;
+  private Thread driverThread = null;
+  private int jobPort = 55555;
+
+  public LyftFlinkPipelineRunner(String driverCmd) {
+    this.driverCmd = driverCmd;
+  }
+
+  /** Main method to be called within the Flink OptimizerPlanEnvironment. */
+  public static void main(String[] args) throws Exception {
+    Preconditions.checkArgument(
+        ExecutionEnvironment.getExecutionEnvironment() instanceof OptimizerPlanEnvironment,
+        "Can only execute in OptimizerPlanEnvironment");
+    LOG.info("entry points args: %s", Arrays.asList(args));
+    EntryPointConfiguration configuration = parseArgs(args);
+    LyftFlinkPipelineRunner runner = new LyftFlinkPipelineRunner(configuration.driverCmd);
+    try {
+      runner.startJobService();
+      runner.runDriverProgram();
+    } catch (Exception e) {
+      throw new RuntimeException(String.format("Job %s failed.", configuration.driverCmd), e);
+    } finally {
+      LOG.info("Stopping job service");
+      runner.stopJobService();
+    }
+    LOG.info("Job submitted successfully.");
+  }
+
+  private static class EntryPointConfiguration {
+    @Option(
+        name = "--driver-cmd",
+        required = true,
+        usage =
+            "Command that launches the Python driver program. "
+                + "(The job service endpoint will be appended as --job_endpoint=localhost:<port>.)")
+    private String driverCmd = null;
```

Comment:
```suggestion
    private String driverCmd;
```

Null is the default for fields.